### PR TITLE
upgraded to Solr 8.6.0

### PR DESCRIPTION
**Pull request by radu-gheorghe**
Adding a PR with the initial commit. It might need more work. Main changes:

- TermsQuery is replaced with TermInSetQuery
- we treat boosts <=0 as Term queries
- upgraded some dependencies (e.g. maven-shade) and using Java 11

### CODE REVIEW

**Comment by prog8** [comment]
I think you'll have to adjust travis configuration because travis builds are failing. I guess you have to remove builds on jdk8 and add builds on jdk11 and newer.

File: src/test/java/com/sematext/solr/redis/TestRedisQParserPluginIT.java [hunk 1]
```diff
@@ -7,23 +7,34 @@
...
   private Jedis jedis;
+  
+  private static RedisServer redisServer;
 
   @BeforeClass
   public static void beforeClass() throws Exception {
     initCore("solrconfig.xml", "schema.xml");
+
+    redisServer = RedisServer.builder()
+    		.port(6379)
```
Comment (prog8) [inline src/test/java/com/sematext/solr/redis/TestRedisQParserPluginIT.java:34]: I think you used tabs and I am pretty sure in all places there are spaces.

File: src/test/java/com/sematext/solr/redis/TestRedisQParserPluginIT.java [hunk 2]
```diff
@@ -1313,5 +1324,11 @@ public void tearDown() throws Exception {
     } catch (final RuntimeException ignored) {
     }
   }
+  
+  @AfterClass
+  public static void afterClass()
```
Comment (prog8) [inline src/test/java/com/sematext/solr/redis/TestRedisQParserPluginIT.java:1329]: `public static void afterClass() {`

File: src/test/java/com/sematext/solr/highlighter/TestTaggedQueryHighlighterIT.java [hunk 3]
```diff
@@ -18,12 +20,20 @@
...
+  private static RedisServer redisServer;
 
   @BeforeClass
   public static void beforeClass() throws Exception {
     initCore("tagged-highlighting-solrconfig.xml", "schema.xml");
     SolrHighlighter highlighter = HighlightComponent.getHighlighter(h.getCore());
     assertTrue("wrong highlighter: " + highlighter.getClass(), highlighter instanceof TaggedQueryHighlighter);
+    
+    redisServer = RedisServer.builder()
+    		.port(6379)
```
Comment (prog8) [inline src/test/java/com/sematext/solr/highlighter/TestTaggedQueryHighlighterIT.java:33]: tabs

File: src/test/java/com/sematext/solr/highlighter/TestTaggedQueryHighlighterIT.java [hunk 4]
```diff
@@ -177,4 +187,10 @@ public void testConstantScoreQueryWithFilterPartOnly() {
     params.add("hl.fl", "location");
     assertQ(req(params), "*[count(//doc)=1]", "count(//lst[@name='highlighting']/*)=1");
   }
+  
+  @AfterClass
+  public static void afterClass()
```
Comment (prog8) [inline src/test/java/com/sematext/solr/highlighter/TestTaggedQueryHighlighterIT.java:192]: ` public static void afterClass() {`

**Comment by otisg** [comment]
@radu-gheorghe doesn't this need to be merged and then a new release made?

File: src/test/java/com/sematext/solr/redis/TestRedisQParserPluginIT.java (hunk 1 above)
Comment (radu-gheorghe) [reply src/test/java/com/sematext/solr/redis/TestRedisQParserPluginIT.java]: Oh, right. I didn't notice that. It should be fixed everywhere now.

File: src/test/java/com/sematext/solr/redis/TestRedisQParserPluginIT.java (hunk 2 above)
Comment (radu-gheorghe) [reply src/test/java/com/sematext/solr/redis/TestRedisQParserPluginIT.java]: Right, fixed.

**Comment by radu-gheorghe** [comment]
Right. I just merged now. I'll do a release right away.

**Pull request merged**